### PR TITLE
[AKS] Check addon profile if it exists

### PR DESCRIPTION
**Pull request by ZeroMagic**
The addon profile is null when there is no addon added before. Here need to check it in the update scenario. 
Ref: #3248 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1736,8 +1736,20 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
...
 
-    azure_keyvault_secrets_provider_addon_profile = instance.addon_profiles.get(
-        CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME, None)
+    azure_keyvault_secrets_provider_addon_profile = None
+    monitoring_addon_enabled = False
+    ingress_appgw_addon_enabled = False
+    virtual_node_addon_enabled = False
+
+    if instance.addon_profiles is not None:
+        azure_keyvault_secrets_provider_addon_profile = instance.addon_profiles.get(CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME, None)
```
[the remark this answers is not in the document]
Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py:1745]: Here I want to return a CLI error. So I separated them.

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -1736,8 +1736,20 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
...
+        monitoring_addon_enabled = CONST_MONITORING_ADDON_NAME in instance.addon_profiles and \
+            instance.addon_profiles[CONST_MONITORING_ADDON_NAME].enabled
+        ingress_appgw_addon_enabled = CONST_INGRESS_APPGW_ADDON_NAME in instance.addon_profiles and \
+            instance.addon_profiles[CONST_INGRESS_APPGW_ADDON_NAME].enabled
+        virtual_node_addon_enabled = CONST_VIRTUAL_NODE_ADDON_NAME + 'Linux' in instance.addon_profiles and \
+            instance.addon_profiles[CONST_VIRTUAL_NODE_ADDON_NAME + 'Linux'].enabled
+
     if enable_secret_rotation:
```
[the remark this answers is not in the document]
Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py:1753]: Currently not. There is no unit test for this file.

File: src/aks-preview/azext_aks_preview/custom.py [hunk 3]
```diff
@@ -1736,8 +1736,20 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
...
         if azure_keyvault_secrets_provider_addon_profile is None or not azure_keyvault_secrets_provider_addon_profile.enabled:
             raise CLIError(
```
[the remark this answers is not in the document]
Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py:1755]: They are in different sub-commonds. Here we use update command to change the property `enableSecretRotation`. As for enabling and disabling the addon, they are located at `az aks enable-addons` and `az aks disable-addons`.

File: src/aks-preview/azext_aks_preview/custom.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (andyliuliming) [reply src/aks-preview/azext_aks_preview/custom.py:1745]: why "return cli error" have relation how you write the code?

[the remark this answers is not in the document]
Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py:1745]: Updated. Make the logic more clear.

**Review by andyliuliming — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**